File: isma-intg-api/src/main/java/ru/nstu/isma/intg/api/IntgResultPoint.java
package ru.nstu.isma.intg.api;

import ru.nstu.isma.intg.api.calcmodel.DaeSystem;

import java.util.Arrays;

/**
 * @author Maria
 * @since 28.03.2016
 */
public class IntgResultPoint {

    private final double x;
    private final double[] yForDe;
    private final double[][] rhs; // TODO: можно сохранять только алгебраические

    public IntgResultPoint(double x, double[] yForDe, double[][] rhs) {
        this.x = x;
        this.yForDe = yForDe;
        this.rhs = rhs;
    }

    public double getX() {
        return x;
    }

    public double[] getYForDe() {
        return yForDe;
    }

    public double[][] getRhs() {
        return rhs;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        IntgResultPoint that = (IntgResultPoint) o;

        if (Double.compare(that.x, x) != 0) return false;
        if (!Arrays.equals(yForDe, that.yForDe)) return false;
        return Arrays.equals(rhs[DaeSystem.RHS_AE_PART_IDX], that.rhs[DaeSystem.RHS_AE_PART_IDX]);
    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        temp = Double.doubleToLongBits(x);
        result = (int) (temp ^ (temp >>> 32));
        result = 31 * result + Arrays.hashCode(yForDe);
        result = 31 * result + Arrays.hashCode(rhs[DaeSystem.RHS_AE_PART_IDX]);
        return result;
    }
}

File: isma-intg-api/src/main/kotlin/ru/nstu/isma/intg/api/models/IntgResultPoint.kt
package ru.nstu.isma.intg.api.models

import ru.nstu.isma.intg.api.calcmodel.DaeSystem

/**
 * @author Maria
 * @since 28.03.2016
 */
class IntgResultPoint(
    val x: Double,
    val yForDe: DoubleArray,
    val rhs: Array<DoubleArray>, // TODO: можно сохранять только алгебраические
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as IntgResultPoint
        if (that.x.compareTo(x) != 0) return false
        return if (!yForDe.contentEquals(that.yForDe)) false else rhs[DaeSystem.RHS_AE_PART_IDX].contentEquals(that.rhs[DaeSystem.RHS_AE_PART_IDX])
    }

    override fun hashCode(): Int {
        var result: Int
        val temp: Long = java.lang.Double.doubleToLongBits(x)

        result = (temp xor (temp ushr 32)).toInt()
        result = 31 * result + yForDe.contentHashCode()
        result = 31 * result + rhs[DaeSystem.RHS_AE_PART_IDX].contentHashCode()
        return result
    }
}